openmetadata-clients/openmetadata-java-client/src/main/java/org/openmetadata/client/security/GoogleAuthenticationProvider.java

```diff
@@ -0,0 +1,75 @@
+package org.openmetadata.client.security;
+
+
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import feign.RequestTemplate;
+import org.openmetadata.catalog.security.client.GoogleSSOClientConfig;
+import org.openmetadata.catalog.services.connections.metadata.OpenMetadataServerConnection;
+import org.openmetadata.client.security.interfaces.AuthenticationProvider;
+
+import java.io.FileInputStream;
+import java.util.Arrays;
+
+public class GoogleAuthenticationProvider  implements AuthenticationProvider {
+    private OpenMetadataServerConnection serverConfig;
+    private GoogleSSOClientConfig securityConfig;
+    private String generatedAuthToken;
+    private Long expirationTimeMillis;
+    private final String OPENID_SCOPE = "https://www.googleapis.com/auth/plus.me";
+    private final String PROFILE_SCOPE = "https://www.googleapis.com/auth/userinfo.profile";
+    private final String EMAIL_SCOPE = "https://www.googleapis.com/auth/userinfo.email";
+
+    public GoogleAuthenticationProvider(OpenMetadataServerConnection iConfig){
+        if(!iConfig.getAuthProvider().equals(OpenMetadataServerConnection.AuthProvider.GOOGLE)){
+            throw new RuntimeException("Required type to invoke is Google OKTA");
+        }
+        serverConfig = iConfig;
+        securityConfig = (GoogleSSOClientConfig) iConfig.getSecurityConfig();
+        generatedAuthToken = "";
+    }
+
+    @Override
+    public AuthenticationProvider create(OpenMetadataServerConnection iConfig) {
+        return new GoogleAuthenticationProvider(iConfig);
+    }
+
+    @Override
+    public String authToken() {
+        try {
+            String credPath = securityConfig.getSecretKey();
```

Comment:
@mohitdeuex we need a way to renew the token as well.

Reply:
@harshach  we refresh it :) 
See this method :-

public void apply(RequestTemplate requestTemplate) {
        if (requestTemplate.headers().containsKey("Authorization")) {
            return;
        }
        // If first time, get the token
        if (expirationTimeMillis == null || System.currentTimeMillis() >= expirationTimeMillis) {
            this.authToken();
        }
        if (getAccessToken() != null) {
            requestTemplate.header("Authorization", "Bearer " + getAccessToken());
        }
    }

Reply:
@mohitdeuex this can run into issues. We need a scheduled thread to keep checking and renewing before the expiry time. 
Imagine this client is embedded in to Trino as event listener the client gets invoked when there is query or table gets created. So there could be long enough duration between the clients and the token can get expired in between.